### PR TITLE
Add tests for animated_switcher.0.dart API example.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Adds tests for `examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart`

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart [hunk 1]
```diff
@@ -11,19 +11,27 @@ void main() => runApp(const AnimatedSwitcherExampleApp());
...
+      home: AnimatedSwitcherExample(duration: duration),
     );
   }
 }
 
 class AnimatedSwitcherExample extends StatefulWidget {
-  const AnimatedSwitcherExample({super.key});
+  const AnimatedSwitcherExample({
+    required this.duration,
+    super.key,
```
Comment (TahaTesser) [inline examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart:27]:
```suggestion
    super.key,
    required this.duration,
```

File: examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart [hunk 2]
```diff
@@ -11,19 +11,27 @@ void main() => runApp(const AnimatedSwitcherExampleApp());
 class AnimatedSwitcherExampleApp extends StatelessWidget {
   const AnimatedSwitcherExampleApp({super.key});
 
+  static const Duration duration = Duration(microseconds: 500);
```
Comment (TahaTesser) [inline examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart:14]: Is there a reason we need static duration?

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart (hunk 2 above)
Comment (ksokolovskyi) [reply examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart:14]: @TahaTesser thanks for the review!
I made the duration static const to reference it in the test.

Comment (TahaTesser) [reply examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart:14]: I don't think we should add unnecessary static variables just for the tests. 

Please revert and hard code duration in the test.

File: examples/api/test/widgets/animated_switcher/animated_switcher.0_test.dart [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    zeroTransition = tester.widget(zeroTransitionFinder);
+    expect(zeroTransition.scale.value, equals(1.0));
+
+    expect(oneTransitionFinder, findsOneWidget);
+    ScaleTransition oneTransition = tester.widget(oneTransitionFinder);
+    expect(oneTransition.scale.value, equals(0.0));
+
+    // Advance animation to the middle.
+    await tester.pump(
+      example.AnimatedSwitcherExampleApp.duration ~/ 2,
```
Comment (TahaTesser) [inline examples/api/test/widgets/animated_switcher/animated_switcher.0_test.dart:76]: If you want pump one half duration, please simply put 250ms value here in single

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart [hunk 4]
```diff
@@ -37,7 +38,7 @@ class _AnimatedSwitcherExampleState extends State<AnimatedSwitcherExample> {
         mainAxisAlignment: MainAxisAlignment.center,
         children: <Widget>[
           AnimatedSwitcher(
-            duration: const Duration(milliseconds: 500),
+            duration: const Duration(microseconds: 500),
```
Comment (TahaTesser) [inline examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart:41]:
```suggestion
            duration: const Duration(milliseconds: 500),
```

microseconds is too fast.

File: examples/api/test/widgets/animated_switcher/animated_switcher.0_test.dart [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    }
+  });
+
+  testWidgets('Animates counter change', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.AnimatedSwitcherExampleApp(),
+    );
+
+    // The animation duration defined in the example app.
+    const Duration animationDuration = Duration(microseconds: 500);
```
Comment (TahaTesser) [inline examples/api/test/widgets/animated_switcher/animated_switcher.0_test.dart:43]:
```suggestion
    const Duration animationDuration = Duration(milliseconds: 500);
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart (hunk 4 above)
Comment (ksokolovskyi) [reply examples/api/lib/widgets/animated_switcher/animated_switcher.0.dart]: @TahaTesser oh, I didn't notice this, thanks!

**Review by TahaTesser — APPROVED** [review]
LGTM

**Review by bleroux — APPROVED** [review]
LGTM!

**Comment by ksokolovskyi** [comment]
@TahaTesser @bleroux thanks for the review!

**Pull request merged**